### ISSUE TITLE
How to simply detect my own scene?

### ISSUE THREAD

**Issue by Rorsc4achhh** [lyclyc52/NeRF_RPN#12]
Just wondering how to simply detect a scene, since in run_rpn_detect.py, there's no detect mode. 
And also for the input data, I got the features file ready, but need some help on preparing the DATASET_SPLIT file since it seems neccesary in the code.
/NeRF_RPN$ bash detect_test.sh
usage: run_rpn_detect.py [-h] [--mode {train,eval}] [--debug_mode]
                         [--features_path FEATURES_PATH] [--boxes_path BOXES_PATH]
                         [--rois_path ROIS_PATH] [--save_root SAVE_ROOT]
                         [--save_path SAVE_PATH] [--dataset_split DATASET_SPLIT]
                         [--checkpoint CHECKPOINT] [--pretrained PRETRAINED]
                         [--bash_file BASH_FILE] [--fine_tune]
                         [--backbone_type {resnet,vgg_AF,vgg_EF,swin}]
                         [--backbone_input_dim BACKBONE_INPUT_DIM]
                         [--resolution RESOLUTION] [--normalize_density]
                         [--output_proposals] [--filter {none,tp,fp}]
                         [--filter_threshold FILTER_THRESHOLD]
                         [--batch_size BATCH_SIZE] [--num_epochs NUM_EPOCHS]
                         [--lr LR] [--reg_loss_weight REG_LOSS_WEIGHT]
                         [--weight_decay WEIGHT_DECAY]
                         [--clip_grad_norm CLIP_GRAD_NORM]
                         [--rotate_prob ROTATE_PROB] [--flip_prob FLIP_PROB]
                         [--rot_scale_prob ROT_SCALE_PROB]
                         [--log_interval LOG_INTERVAL]
                         [--eval_interval EVAL_INTERVAL]
                         [--keep_checkpoints KEEP_CHECKPOINTS] [--wandb]
                         [--process_root PROCESS_ROOT]
                         [--process_name PROCESS_NAME] [--init_method INIT_METHOD]
                         [--gpus GPUS] [--n_classes N_CLASSES]
                         [--output_size OUTPUT_SIZE [OUTPUT_SIZE ...]]
                         [--spatial_scale SPATIAL_SCALE [SPATIAL_SCALE ...]]
                         [--feature_input_dim FEATURE_INPUT_DIM] [--obj_only]
                         [--enlarge_scale ENLARGE_SCALE] [--use_cuda] [--remap]
                         [--is_add_layer]
                         [--feature_extracting_type {pooling,interpolation}]
                         [--nms_thresh NMS_THRESH]
                         [--filter_score_threhold FILTER_SCORE_THREHOLD]
                         [--filter_num_threhold FILTER_NUM_THREHOLD]
                         [--cls_batch_size CLS_BATCH_SIZE]
                         [--fg_fraction FG_FRACTION] [--fg_threshold FG_THRESHOLD]
                         [--bg_threshold BG_THRESHOLD] [--top_k TOP_K]
                         [--rotated_bbox] [--is_flatten] [--log_to_file]
                         [--output_all]

**Comment by zymk9** [comment]
Hi. Sorry for the confusion. The code in `run_rpn_detect.py` is mainly used to carry out ablation study on whether adding a binary classification R-CNN head can improve the region proposals. That's why you will see that `--n_classes` is fixed to 2, and in `RPNClassificationDataset` under `datasets.py` we do not load class labels other than using foreground/background, since the code is for binary classification only.

That said, we have a follow up work which performs 3D multi-object detection as well as instance segmentation in NeRF. It contains a "more complete" version of `run_rpn_detect.py`. The repo is [here](https://github.com/lyclyc52/Instance_NeRF) and we plan to release to code before October.

**Comment by Rorsc4achhh** [comment]
Thanks for your reply! Looking forward to your future work.
